File: Brute-force/Permutation/permutation.java
    private static void permutation(boolean[] visited, int[] arr, int cnt) {
        if (cnt == N) {
            System.out.println(Arrays.toString(arr));
            return;
        }
        for (int i = 1; i <= N; ++i) {
            if (!visited[i]) {
                visited[i] = true;
                arr[cnt] = i;
                permutation(visited, arr, cnt + 1);
                visited[i] = false;
            }
        }
    }

File: Brute-force/Permutation/permutation.py
def permutation(arr, visited, cnt):
    if cnt == N:
        print(*arr)
        return
    for i in range(1, N + 1):
        if not visited[i]:
            visited[i] = True
            arr[cnt] = i
            permutation(arr, visited, cnt + 1)
            visited[i] = False
